apollo-router-core/src/spec/field_type.rs

```diff
@@ -0,0 +1,106 @@
+use crate::*;
+use apollo_parser::ast;
+
+// Primitives are taken from scalars: https://spec.graphql.org/draft/#sec-Scalars
+#[derive(Debug)]
+pub(crate) enum FieldType {
+    Named(String),
+    List(Box<FieldType>),
+    NonNull(Box<FieldType>),
+    String,
+    Int,
+    Float,
+    Id,
+    Boolean,
+}
+
+impl FieldType {
+    pub(crate) fn validate_value(&self, value: &Value, schema: &Schema) -> bool {
+        match (self, value) {
+            (FieldType::String, Value::String(_)) => true,
+            (FieldType::Int, Value::Number(number)) if !number.is_f64() => true,
```

Comment:
why not `is_i64() || is_u64()` here? The documentation for `is_f64()` states that it would be equivalent but puts no guarantee on it: https://docs.serde.rs/serde_json/struct.Number.html#method.is_f64
(there are cases where I could see this failing, like a floating point number that requires 128 bits, because 64 bits is not required by the JSON spec)

Reply:
I didn't think much about this. I'll fix it

Reply:
Fixed in e42d107

I kept `is_f64()` for the float type and I use `is_i64() || is_u64()` for the integer type. I think that makes more sense right?

Reply:
:+1: